bcgov/HOUS-permit-portal
# Feature/bphh 635/contact requirement types

## Description
- Adds "General Contact" and "Professional Contact" requirement types
- Adds ability to configure the above from super admin side
- Adds the ability to render above requirements in the submitter flow
- Also refactors internal code fort better organization


## What type of PR is this? (check all applicable)

- [X ] 🍕 Feature

## Related Tickets & Documents
https://hous-bssb.atlassian.net/browse/BPHH-635


## Steps to QA
- Super admin config
  - Log in as Super admin
  - Navigate to the requirements library and select a requirement block to edit
  - The new contact requirements should be available to add from the drawer and configurable
  - After adding, they should be visible in all requirement templates which use them
- Login as submitter
  - Select one draft permit (this is already seeded)
  - You should be able to see contact requirement types and fill them
  - They should persist on save
  - The multi addable contacts is not included in seeds so is not possible to test without manual changes
